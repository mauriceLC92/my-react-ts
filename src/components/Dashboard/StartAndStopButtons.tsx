import * as React from 'react';


interface State {

}

interface Props {
    onStart: Function;
    onStop: Function;
}
export class StartAndStopButtons extends React.Component <Props, State>{
    constructor(props: Props){
        super(props);

        this.startRace = this.startRace.bind(this);
        this.resetRace = this.resetRace.bind(this);
    }

    startRace = () =>{
        // this.setState({run: true})
        this.props.onStart(true)
    }

    resetRace = () =>{
        // this.setState({run: true})
        this.props.onStop(false)
    }
    
    render(){
        return(
            <div className="ui medium buttons">
                  <button id="startButton" className="positive ui button" onClick={this.startRace}>Render Race Tracks</button>
                  <div className="or"></div>
                  <button className="negative ui button" onClick={this.resetRace}>Clear Race Tracks</button>
            </div>
        );
    }
}