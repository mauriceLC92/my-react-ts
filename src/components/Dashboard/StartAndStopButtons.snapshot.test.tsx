import * as React from 'react';
const renderer =require('react-test-renderer');
import {StartAndStopButtons} from './StartAndStopButtons';

describe('StartAnsStopbuttons component renders the startandstop correctly', () => {
    it('renders correctly', () => {
      
      const rendered = renderer.create(
        <StartAndStopButtons />
      );
      expect(rendered.toJSON()).toMatchSnapshot();
    });


    it('should render racetracks', () => {
        const component = renderer.create(<StartAndStopButtons />);
        component.getInstance().startRace();
        expect(component.toJSON()).toMatchSnapshot();
      });
  });
