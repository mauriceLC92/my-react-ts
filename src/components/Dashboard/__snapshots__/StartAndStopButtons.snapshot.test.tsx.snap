// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`StartAnsStopbuttons component renders the startandstop correctly renders correctly 1`] = `
<div
  className="ui medium buttons"
>
  <button
    className="positive ui button"
    id="startButton"
    onClick={[Function]}
  >
    Render Race Tracks
  </button>
  <div
    className="or"
  >
    .
  </div>
  <button
    className="negative ui button"
    onClick={[Function]}
  >
    Clear Race Tracks
  </button>
</div>
`;
